src/components/EditCoursePage/index.jsx

```diff
@@ -0,0 +1,213 @@
+    courseData.video = courseData.videoSrc;
+    courseData.uuid = uuid;
+    return this.props.editCourse(courseData);
+  }
+
+  render() {
+    if (!this.props.courseInfo || !this.props.courseOptions) {
+      return (
+        <StatusAlert
+          id="error"
+          alertType="danger"
+          title="Course Edit Form failed to load: "
+          message="Course information unavailable."
+        />
+      );
+    }
+
+    const {
+      courseInfo,
+      courseInfo: {
+        data: {
+          title,
+          key,
+          short_description,
+          full_description,
+          outcome,
+          subjects,
+          image,
+          prerequisites_raw,
+          level_type,
+          learner_testimonials,
+          faq,
+          additional_information,
+          syllabus_raw,
+          video,
+          entitlements,
+        },
+      },
+      courseOptions,
+    } = this.props;
+
+    // If we want to keep a lot of the logic in the lower return,
+    // we have to do all this weird '&&' logic to make sure the data is there.
+    const keyParts = key && key.split(/\+|\//);
+    const number = keyParts && keyParts[keyParts.length - 1];
+    const subjectMap = subjects && subjects.map(x => x.slug);
+    const subjectPrimary = subjectMap && subjectMap[0];
+    const subjectSecondary = subjectMap && subjectMap[1];
+    const subjectTertiary = subjectMap && subjectMap[2];
+    const imageSrc = image && image.src;
+    const videoSrc = video && video.src;
+    const entitlement = entitlements && entitlements[0];
+    const mode = (entitlement && entitlement.mode) || 'Audit';
+    const price = entitlement && entitlement.price;
+
+    return (
+      <React.Fragment>
+        <Helmet>
+          <title>{`Course - ${title}`}</title>
+        </Helmet>
+
+        <div className="container-fluid">
+          <div className="row justify-content-md-center my-3 ">
+            <div className="col-6">
+              { (!this.state.startedFetching || courseInfo.isFetching ||
+                courseOptions.isFetching) &&
+                <LoadingSpinner /> }
+              { courseInfo.error && (
+                <StatusAlert
+                  id="error"
+                  alertType="danger"
+                  title="Course Edit Form failed to load: "
+                  message={courseInfo.error}
+                />
+              )}
+              { courseOptions.error && (
+                <StatusAlert
+                  id="error"
+                  alertType="danger"
+                  title="Course Edit Form failed to load: "
+                  message={courseOptions.error}
+                />
+              )}
+              { (this.state.startedFetching && !courseInfo.isFetching &&
+                !courseOptions.isFetching) && (
+                <div>
+                  <h2>Course</h2>
+                  <hr />
+                  <div className="col">
+                    <EditCourseForm
```

Comment:
The alignment of this component inside the broader component isn't quite right. The left and right edges don't line up with the "Course" title and the "Save Course" button. Doesn't need to be done here - we can beautfy later. But just mentioning.